velox/functions/prestosql/Comparisons.cpp

```diff
@@ -52,7 +52,7 @@ struct SimdComparator {
     if constexpr (numScalarElements == 2 || numScalarElements == 4) {
       for (auto i = begin; i < vectorEnd; i += 8) {
         rawResult[i / 8] = 0;
-        for (auto j = 0; j < 8 && j < vectorEnd; j += numScalarElements) {
+        for (auto j = 0; j < 8 && (i + j) < vectorEnd; j += numScalarElements) {
```

Comment:
Can you add a test case that exercises this fix ?

Reply:
Constructing this case is very challenging because if the address following the effective address are valid(memory allocated elsewhere, which cannot be controlled), it's impossible to trigger a crash issue, and the calculated results would also be correct.Successfully running the existing cases can prove that the fix is effective.

Reply:
One way to test this out is to construct two flat vectors of the right type (maybe its double / bigint , I will have to double check ) that will exercise this condition . You can custom create the buffer of values and have it of size larger than the size of the vector (Say if vector is size of 10, then have the buffer large enough to have 12 values). Then ensure the last two values in the buffer are set with values that are different in the vector and do a simple neq comparison and check if the last two bits are set in the result vector.

Reply:
I have added the test cases following the method you provided. Thanks！

Reply:
Thanks !